distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso

```diff
@@ -619,3 +704,27 @@ truncate_table_truncate_table_style dialect table_name =
 default_fetch_types_query dialect expression context where_filter_always_false_literal="FALSE" =
     empty_context = context.add_where_filters [SQL_Expression.Literal where_filter_always_false_literal]
     dialect.generate_sql (Query.Select [["typed_column", expression]] empty_context)
+
+## PRIVATE
+   Helper class for shortening the binder names generated for WITH clauses.
+type Let_Binder_Renamer
+    Value (existing_tables : Hashset Text) (binders_ref : Ref Dictionary) (serial_ref : Ref Integer)
+
+    new (existing_tables : Vector Text) -> Let_Binder_Renamer =
+        Let_Binder_Renamer.Value (Hashset.from_vector existing_tables) (Ref.new Dictionary.empty) (Ref.new 0)
+
+    rename self (name : Text) (binder : Text) =
+        binders = self.binders_ref.get
+        if binders.contains_key binder then binders.get binder else
```

Comment:
nitpick as that's probably not a bottleneck, but what if we `binders.get binder if_missing=Nothing`? We could avoid accessing the hashmap twice 😅

Reply:
Done 😅